request.go

```diff
@@ -27,13 +27,24 @@ var (
 	// (numeric) but the Struct field was a non numeric type (i.e. not int, uint,
 	// float, etc)
 	ErrUnknownFieldNumberType = errors.New("The struct field was not of a known number type")
-	// ErrUnsupportedPtrType is returned when the Struct field was a pointer but
-	// the JSON value was of a different type
-	ErrUnsupportedPtrType = errors.New("Pointer type in struct is not supported")
 	// ErrInvalidType is returned when the given type is incompatible with the expected type.
 	ErrInvalidType = errors.New("Invalid type provided") // I wish we used punctuation.
 )
 
+// ErrUnsupportedPtrType is returned when the Struct field was a pointer but
+// the JSON value was of a different type
+func ErrUnsupportedPtrType(rf reflect.Value, t reflect.Type, structField reflect.StructField) error {
```

Comment:
Please refactor this so as to use the `error` interface, the standard Go pattern for errors that have state.

1) Create a **public/exported** `struct` called called `ErrUnsupportedPtrType` with private (lower-case) fields `rf`, `t`, and `structField` as members.
2) Implement `func (eupt ErrUnsupportedPtrType) Error() string` to complete the `error` interface, See, https://golang.org/src/builtin/builtin.go?h=error#L255.
3) Create a **private/unexported function** called newErrUnsupportedPtrType and take the formal params that are listed above, `reflect.Value`, `reflect.Type`, `reflect.StructField` and returns a `ErrUnsupportedPtrType`, for convenience.
4) Use `return newErrUnsupportedPtrType(...)` in place of your exported function here.